### PR TITLE
Did the first task of issue #42016: removing the default address (https://localhost:5001) for https

### PR DESCRIPTION
**Pull request by Daniel-Genkin-MS-2**
This PR removes the default fallback address for HTTPS connections. This is part of step 1 of Damian's issue #42016. 

I also modified the tests that broke to remove the references to https. As they specifically tested for defaults, I decided not to replace the https parts. I am not sure though if this has put holes in the test coverage. I don't believe they did but it is probably worth keeping in mind for those reviewing the PR.

Contributes to: #42016

### CODE REVIEW

File: src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs [hunk 1]
```diff
@@ -151,21 +151,8 @@ public async Task BindAsync(AddressBindContext context, CancellationToken cancel
...
-                await httpsDefault.BindAsync(context, cancellationToken).ConfigureAwait(false);
-                context.Logger.LogDebug(CoreStrings.BindingToDefaultAddresses,
-                    Constants.DefaultServerAddress, Constants.DefaultServerHttpsAddress);
-            }
-            else
-            {
-                // No default cert is available, do not bind to the https endpoint.
-                context.Logger.LogDebug(CoreStrings.BindingToDefaultAddress, Constants.DefaultServerAddress);
-            }
+            // No default cert is available, do not bind to the https endpoint.
```
Comment (Tratcher) [inline src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs:154]:
```suggestion
```

File: src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs [hunk 2]
```diff
@@ -151,21 +151,8 @@ public async Task BindAsync(AddressBindContext context, CancellationToken cancel
...
             await httpDefault.BindAsync(context, cancellationToken).ConfigureAwait(false);
 
-            // Conditional https default, only if a cert is available
-            var httpsDefault = ParseAddress(Constants.DefaultServerHttpsAddress, out _);
-            context.ServerOptions.ApplyEndpointDefaults(httpsDefault);
-
-            if (httpsDefault.IsTls || httpsDefault.TryUseHttps())
-            {
-                await httpsDefault.BindAsync(context, cancellationToken).ConfigureAwait(false);
-                context.Logger.LogDebug(CoreStrings.BindingToDefaultAddresses,
```
Comment (Tratcher) [inline src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs:161]: https://github.com/dotnet/aspnetcore/blob/d96a100bddc72606f7417b665428411388b8ac54/src/Servers/Kestrel/Core/src/CoreStrings.resx#L426
This message is probably not used anymore and can be removed.

File: src/Servers/Kestrel/test/BindTests/AddressRegistrationTests.cs [hunk 3]
```diff
@@ -522,13 +519,8 @@ private async Task RegisterDefaultServerAddresses_Success(IEnumerable<string> ad
...
             Assert.Equal(5000, host.GetPort());
 
-            if (mockHttps)
-            {
-                Assert.Contains(5001, host.GetPorts());
-            }
-
             Assert.Single(LogMessages, log => log.LogLevel == LogLevel.Debug &&
-                (string.Equals(CoreStrings.FormatBindingToDefaultAddresses(Constants.DefaultServerAddress, Constants.DefaultServerHttpsAddress), log.Message, StringComparison.Ordinal)
+                (string.Equals(CoreStrings.FormatBindingToDefaultAddresses(Constants.DefaultServerAddress, "https://localhost:0"), log.Message, StringComparison.Ordinal)
```
Comment (Tratcher) [inline src/Servers/Kestrel/test/BindTests/AddressRegistrationTests.cs:523]:
```suggestion
                (string.Equals(CoreStrings.FormatBindingToDefaultAddress(Constants.DefaultServerAddress), log.Message, StringComparison.Ordinal)
```

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Since you'll need a breaking change announcement for this, you can draft that here as a PR comment for us to review.
See the template at https://github.com/aspnet/Announcements/issues/new?assignees=&labels=Breaking+change&template=breaking-change.yaml&title=%5BBreaking+change%5D%3A+

**Review by halter73 — APPROVED** [review]

**Comment by Daniel-Genkin-MS-2** [comment]
Here is my first draft for the announcement. Please let me know if there is anything that can be improved. Namely, I link the .NET 6 guide whereas this is an announcement for .NET 7. I didn't see an option for .NET 7 in the version selector. Is this ok? Also, are all my suggestions in the new behavior still current? Should I remove some?

# Announcement draft

## Description
The default HTTPS address and port have been removed from Kestrel in .NET 7 preview 5. This is step one in https://github.com/dotnet/aspnetcore/issues/42016 which will improve overall developer experience when dealing with HTTPS.

PR that makes the change: https://github.com/dotnet/aspnetcore/pull/42021

## Version
.NET 7 preview 5

## Previous Behavior
Previously, if no values for the address and port were specified explicitly but a local development certificate was available, Kestrel would default to binding to https://localhost:5001.

## New Behaviour
Users must now manually bind to HTTPS and specify the address and port explicitly, either via the `launchSettings.json` file, the `ASPNETCORE_URLS` environment variable, `--urls` command line argument, `urls` host configuration key, or via the [`UseUrls`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.hostingabstractionswebhostbuilderextensions.useurls?view=aspnetcore-6.0) extension method.

## Type of breaking change
Source incompatibility

## Reason for change
This current eager binding behavior occurs without regard to the configured environment and can lead to experience issues on developer machines when the certificate has not yet been trusted (i.e. trusted as root cert authority because it's self-signed). Clients often produce poor UX when hitting an HTTPS endpoint with an untrusted certificate, e.g. silent failure, scary error/warning screen, etc.

## Recommended action
If you were not using the default https://localhost:5001 binding, no changes are required. However, if you were using this binding, please make sure to review [this guide](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-6.0) on how you can update your server to meet this new behavior.

## Affected APIs
Kestrel Address Binding

**Comment by Daniel-Genkin-MS-2** [comment]
@halter73 Do you have any comments on this announcement?

Also. should we maybe first update the docs somehow before posting the announcement? Or should I just remove the link to the guide as it is for .NET 6?

**Comment by Daniel-Genkin-MS-2** [comment]
@Tratcher I made the changes that I could. Unfortunately, the "Type of change" is a set of 2 check boxes and runtime behavior change is not an option.
image

**Comment by Tratcher** [comment]
> @halter73 Do you have any comments on this announcement?
> 
> Also. should we maybe first update the docs somehow before posting the announcement? Or should I just remove the link to the guide as it is for .NET 6?

Don't wait for the doc contents, we'll get that updated separately. The doc still covers how to enable https manually.

**Pull request merged**